numba_dpex/experimental/kernel_dispatcher.py

```diff
@@ -81,10 +84,18 @@ def _compile_to_spirv(
         kernel_fn = kernel_targetctx.prepare_spir_kernel(
             kernel_func, kernel_fndesc.argtypes
         )
-
-        # makes sure that the spir_func is completely inlined into the
-        # spir_kernel wrapper
-        kernel_library.optimize_final_module()
+        # If the inline_threshold option was set then set the property in the
+        # kernel_library to force inlining ``overload`` calls into a kernel.
+        try:
+            kernel_library.inline_threshold = self.targetoptions[
```

Comment:
```suggestion
            kernel_library.inline_threshold = self.targetoptions.get(
```
Should we use `get` instead of try-catch?

Reply:
Yes, I was not aware of it. Will make the change.

Reply:
Addressed